### PR TITLE
Fix next link for audit show page (take 2) [SEAL IGNORE]

### PR DESCRIPTION
**Pull request by cbaines**
See comment in #198 

Trello: https://trello.com/c/B8YQWsfv/308-possible-bug-if-you-click-the-last-item-on-a-list-page-you-dont-get-a-next-link

### CODE REVIEW

**Comment by tuzz** [comment]
This is all good work but it really needs tests. This is an important feature / bug fix and without tests we increase our risk this will break again in the future.

**Comment by cbaines** [comment]
> This is all good work but it really needs tests. This is an important feature / bug fix and without tests we increase our risk this will break again in the future.

Ok, I'll take a look at improving the testsuite.

File: app/controllers/audits_controller.rb [hunk 1]
```diff
@@ -59,8 +65,34 @@ def content_items
     @content_items ||= search.content_items.decorate
   end
 
-  def next_item
-    @next_item ||= content_items.next_item(content_item)
+  def compute_next_item_and_page
+    # Get a list of ContentItem ids for everything on the current
```
Comment (pmanrubia) [inline app/controllers/audits_controller.rb]: Would you say this belongs in the controller?

Comment (pmanrubia) [inline app/controllers/audits_controller.rb]: What are your thoughts about extracting this calculation into a class? It seems a bit complex to live in the controller.

Something along the lines of:

```ruby
iterator = @search.iterator()
content_item = iterator.next(starting: index)
```

or

```ruby
iterator = Audit::Iterator.new(@search)
content_item = iterator.next()
```

And include all the logic + tests in the `Audit::Iterator` class as a first step to isolate this behaviour?

Comment (cbaines) [inline app/controllers/audits_controller.rb]: This method has two side effects, the `@next_item`, but also the `@next_link_filter_params`, which is like the `filter_params`, but can include a different page.

I did think of having some function that would return the two bits of information, e.g. `next_item, next_page = next_item_and_page(scope, current_content_item)`, but decided against in as just having a method that set instance variables seemed similar to the style of the AuditsController.

I'm not against having it in a class, I put it in the controller for a start to keep the complexity low.

**Comment by pmanrubia** [comment]
I am closing this PR. 
All the valuable findings have been recorded and tracked via this [Trello card](https://trello.com/c/B8YQWsfv/308-if-you-click-the-last-item-on-a-list-page-you-dont-get-a-next-link). We plan to re-implement navigation from scratch to address the issues that we are currently having.

**Pull request closed without merging**